compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -8207,9 +8208,34 @@ private STNode parseCallStatement(STNode expression) {
         // This is not a must because this expression is validated in the semantic analyzer.
         STNode semicolon = parseSemicolon();
         endContext();
+        if (expression.kind == SyntaxKind.CHECK_EXPRESSION) {
+            STNode callExpr = parseCallExpression((STCheckExpressionNode) expression);
+            return STNodeFactory.createExpressionStatementNode(SyntaxKind.CALL_STATEMENT, callExpr, semicolon);
+        }
         return STNodeFactory.createExpressionStatementNode(SyntaxKind.CALL_STATEMENT, expression, semicolon);
     }
 
+    private STNode parseCallExpression(STCheckExpressionNode callExpr) {
+        STNode expr = callExpr.expression;
+        if (expr.kind == SyntaxKind.CHECK_EXPRESSION) {
+            STNode checkExpr = parseCallExpression((STCheckExpressionNode) expr);
+            return STNodeFactory.createCheckExpressionNode(SyntaxKind.CHECK_EXPRESSION, callExpr.checkKeyword,
+                                                           checkExpr);
+        } else if (callExpr.expression.kind != SyntaxKind.FUNCTION_CALL && expr.kind != SyntaxKind.METHOD_CALL) {
+            STNode checkingKeyword = SyntaxErrors.cloneWithTrailingInvalidNodeMinutiae(callExpr.checkKeyword, expr,
+                                                DiagnosticErrorCode.ERROR_INVALID_EXPRESSION_EXPECTED_CALL_EXPRESSION);
+            STNode funcName = SyntaxErrors.createMissingToken(SyntaxKind.IDENTIFIER_TOKEN);
+            funcName = STNodeFactory.createSimpleNameReferenceNode(funcName);
+            STNode openParenToken = SyntaxErrors.createMissingToken(SyntaxKind.OPEN_PAREN_TOKEN);
+            STNode arguments = STNodeFactory.createEmptyNodeList();
+            STNode closeParenToken = SyntaxErrors.createMissingToken(SyntaxKind.CLOSE_PAREN_TOKEN);
+            STNode funcCallExpr = STNodeFactory.createFunctionCallExpressionNode(funcName, openParenToken, arguments,
+                                                                                 closeParenToken);
+            return STNodeFactory.createCheckExpressionNode(SyntaxKind.CHECK_EXPRESSION, checkingKeyword, funcCallExpr);
+        }
+        return callExpr;
```

Comment:
we can simplify this by returning `callExpr` if the `expr` is func call or method call at the beginning of the method